[Bug] EDM4U 1.2.183: Error occurs in GooglePlayGamesPlugin

<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/issue.md
-->

### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2022.3.22f1
  * External Dependency Manager version: 1.2.183
  * Source you installed EDM4U: GoogleMobileAds-v9.3.0.unitypackage, GooglePlayGamesPlugin-0.11.01.unitypackage (.unitypackage or Unity Package Manager)
  * Features in External Dependency Manager in use: Android Resolver (Android Resolver, iOS Resolver, VersionHandler, etc.)
  * Plugins SDK in use: AdMob, play-games-plugin (Firebase, Admob, Facebook, etc.)
  * Platform you are using the Unity editor on: Windows (Mac, Windows, or Linux)

### [REQUIRED] Please describe the issue here:
(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)

※I am Japanese. I am using a translation site for questions. Please let me know if there is any misunderstanding.

①. Create two new projects in Unity: A-project and B-project.
②. A-project: Import GooglePlayGamesPlugin-0.11.01.unitypackage.
③. B-project: Import GoogleMobileAds-v9.3.0.unitypackage.
④. A-project: Delete ExternalDependencyManager.
⑤. A-project: Copy ExternalDependencyManager from B-project to A-project.
⑥. A-project: Run [Assets]-[External Dependency Manager]-[Android Resolver]-[Resolve].
⑦. A-project: Run [Build Setting]-[Build].
The following error occurs.

```
Execution failed for task ':launcher:mergeLibDexDebug'.
> Could not resolve all files for configuration ':launcher:debugRuntimeClasspath'.
   > Failed to transform classes.jar (project :unityLibrary) to match attributes {artifactType=android-dex, asm-transformed-variant=NONE, com.android.build.api.attributes.AgpVersionAttr=7.1.2, com.android.build.api.attributes.BuildTypeAttr=debug, com.android.build.gradle.internal.attributes.VariantAttr=debug, dexing-enable-desugaring=true, dexing-is-debuggable=true, dexing-min-sdk=22, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Could not resolve all files for configuration ':launcher:debugRuntimeClasspath'.
         > No variants of com.google.games:gpgs-plugin-support:0.11.01 match the consumer attributes:
             - com.google.games:gpgs-plugin-support:0.11.01 configuration runtime:
                 - Incompatible because this component declares attribute 'artifactType' with value 'srcaar' and the consumer needed attribute 'artifactType' with value 'android-classes-jar'
                 - Other compatible attributes:
                     - Doesn't say anything about asm-transformed-variant (required 'NONE')
                     - Doesn't say anything about dexing-enable-desugaring (required 'true')
                     - Doesn't say anything about dexing-is-debuggable (required 'true')
                     - Doesn't say anything about dexing-min-sdk (required '22')

```
If you look at \Assets\Plugins\Android\settingsTemplate.gradle, you will see the following:
`    url (unityProjectPath + "/Assets/GooglePlayGames/com.google.play.games/Editor/m2repository") // Assets/GooglePlayGames/com.google.play.games/Editor/GooglePlayGamesPluginDependencies.xml:11
`
In EDM4U 1.2.179, GeneratedLocalRepo is written in the path.
`    url "file:///D:/ApplicationUnity_SDK/GooglePlayGamesPlugin/Assets/GeneratedLocalRepo/GooglePlayGames/com.google.play.games/Editor/m2repository" // Assets/GooglePlayGames/com.google.play.games/Editor/GooglePlayGamesPluginDependencies.xml:11
`
I think the error occurs because GeneratedLocalRepo is not specified. Is that correct?

◇Package download site
https://github.com/googleads/googleads-mobile-unity/releases/tag/v9.3.0
https://github.com/playgameservices/play-games-plugin-for-unity


#### Please answer the following, if applicable:
What's the issue repro rate? (eg 100%, 1/5 etc)

100%

Thank you for your reply.

I tried to build it using the method you instructed, but an error occurred.

1. Create new Project
2. Add GooglePlayGamesPlugin-0.11.01.unitypackage via Assets > Import Package > Custom Package...
3. Delete External Dependency Manager folder under the Assets folder.
4. Add external-dependency-manager-1.2.183.unitypackage via Assets > Import Package > Custom Package...
5. [Assets > External Dependency Manager] is not displayed, so restart the computer.
6. The following dialog will appear. Select Apply.
```
    Google Version Handler
    Would you like to delete the following obsolete files in your project?
    Assets\Play ServicesResolver\Editor\play-services-resolver_v1.2.137.0.txt (GooglePlayGamesPlugin)

```
7. Go to Assets > External Dependency Manager > Android Resolver > Settings, turn on "When building Android app through Unity" under Use Full Custom Local Maven Repo Path
```
*A warning message was displayed.
Assets/GooglePlayGames/com.google.play.games/Editor/GooglePlayGamesPluginDependencies.xml:11: Repo path 'Packages/com.google.play.games/Editor/m2repository' does not exist.

```
8. Run Assets > External Dependency Manager > Android Resolver > Resolve
*Assets/GeneratedLocalRepo was not created.

7. Run [File]-[Build Setting]-[Build]

```
*The following error occurred:

Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'. See the Console for details.

――――――――――――――――――――――――――――――――
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':launcher:checkReleaseAarMetadata'.
> Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
   > Could not find com.google.games:gpgs-plugin-support:0.11.01.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - https://repo.maven.apache.org/maven2/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GooglePlayGamesPlugin_0_11_01/Packages/com.google.play.games/Editor/m2repository/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/C:/Users/nor25/.m2/repository/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GooglePlayGamesPlugin_0_11_01/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/libs/gpgs-plugin-support-0.11.01.jar
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GooglePlayGamesPlugin_0_11_01/Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/libs/gpgs-plugin-support.jar
     Required by:
         project :launcher > project :unityLibrary

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':unityLibrary:generateReleaseRFile'.
> Could not resolve all files for configuration ':unityLibrary:releaseCompileClasspath'.
   > Could not find com.google.games:gpgs-plugin-support:0.11.01.
     Required by:
         project :unityLibrary

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 9s

UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)
```

I fixed the following warning, but an error still occurred.
`Assets/GooglePlayGames/com.google.play.games/Editor/GooglePlayGamesPluginDependencies.xml:11: Repo path 'Packages/com.google.play.games/Editor/m2repository' does not exist.`

1. Warning correction
\Assets\GooglePlayGames\com.google.play.games\Editor\GooglePlayGamesPluginDependencies.xml
Packages/com.google.play.games/Editor/m2repository
↓
Assets/GooglePlayGames/com.google.play.games/Editor/m2repository

2. Run Assets > External Dependency Manager > Android Resolver > Resolve.
*Assets/GeneratedLocalRepo was not created.

3. Run Assets > External Dependency Manager > Android Resolver > Force Resolve.
*Assets/GeneratedLocalRepo has been created.

4. Run [File]-[Build Setting]-[Build]

The following error occurred:
```
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
注意:D:\ApplicationUnity_SDK\Google\_GooglePlayGamesPlugin\GooglePlayGamesPlugin_0_11_01\Library\Bee\Android\Prj\IL2CPP\Gradle\unityLibrary\src\main\java\com\unity3d\player\UnityPlayerActivity.javaは推奨されないAPIを使用またはオーバーライドしています。
注意:詳細は、-Xlint:deprecationオプションを指定して再コンパイルしてください。

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':launcher:mergeDexRelease'.
> Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
   > Failed to transform classes.jar (project :unityLibrary) to match attributes {artifactType=android-dex, asm-transformed-variant=NONE, com.android.build.api.attributes.AgpVersionAttr=7.1.2, com.android.build.api.attributes.BuildTypeAttr=release, com.android.build.gradle.internal.attributes.VariantAttr=release, dexing-enable-desugaring=true, dexing-is-debuggable=false, dexing-min-sdk=22, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
         > No variants of com.google.games:gpgs-plugin-support:0.11.01 match the consumer attributes:
             - com.google.games:gpgs-plugin-support:0.11.01 configuration runtime:
                 - Incompatible because this component declares attribute 'artifactType' with value 'srcaar' and the consumer needed attribute 'artifactType' with value 'android-classes-jar'
                 - Other compatible attributes:
                     - Doesn't say anything about asm-transformed-variant (required 'NONE')
                     - Doesn't say anything about dexing-enable-desugaring (required 'true')
                     - Doesn't say anything about dexing-is-debuggable (required 'false')
                     - Doesn't say anything about dexing-min-sdk (required '22')

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 7s

UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)

```
When I looked at \Assets\Plugins\Android\settingsTemplate.gradle
, it was url "file:///D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GooglePlayGamesPlugin_0_11_01/Assets/GooglePlayGames/com.google.play.games/Editor/m2repository"
and it didn't reference GeneratedLocalRepo.

Best Regards
crSudo

@argzdev 
Thank you for your reply.

■Unity Update Verification
When I imported dependency-manager-1.2.183.unitypackage in Unity 2022.3.51f1, the following error occurred.

I also fixed \Assets\GooglePlayGames\com.google.play.games\Editor\GooglePlayGamesPluginDependencies.xml.

```
> Configure project :launcher
WARNING:The option setting 'android.aapt2FromMavenOverride=D:\Tool\Unity\2022.3.51f1\Editor\Data\PlaybackEngines\AndroidPlayer\SDK\build-tools\34.0.0\aapt2.exe' is experimental.

> Configure project :unityLibrary:GooglePlayGamesManifest.androidlib
WARNING:minSdkVersion (22) is greater than targetSdkVersion (16) for variant "debug". Please change the values such that minSdkVersion is less than or equal to targetSdkVersion.
WARNING:minSdkVersion (22) is greater than targetSdkVersion (16) for variant "release". Please change the values such that minSdkVersion is less than or equal to targetSdkVersion.
WARNING:We recommend using a newer Android Gradle plugin to use compileSdk = 35

This Android Gradle plugin (7.4.2) was tested up to compileSdk = 33

This warning can be suppressed by adding
    android.suppressUnsupportedCompileSdk=35
to this project's gradle.properties

The build will continue, but you are strongly encouraged to update your project to
use a newer Android Gradle Plugin that has been tested with compileSdk = 35

> Task :unityLibrary:preBuild UP-TO-DATE
99 actionable tasks: 50 executed, 49 up-to-date

UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)

```

```
����:D:\ApplicationUnity_SDK\Google\_GooglePlayGamesPlugin\GPG_0_11_01_U2022_3_51\Library\Bee\Android\Prj\IL2CPP\Gradle\unityLibrary\src\main\java\com\unity3d\player\UnityPlayerActivity.java�͐�������Ȃ�API��g�p�܂��̓I�[�o�[���C�h���Ă��܂��B
����:�ڍׂ́A-Xlint:deprecation�I�v�V������w�肵�čăR���p�C�����Ă��������B

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':launcher:mergeDexRelease'.
> Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
   > Failed to transform classes.jar (project :unityLibrary) to match attributes {artifactType=android-dex, asm-transformed-variant=NONE, com.android.build.api.attributes.AgpVersionAttr=7.4.2, com.android.build.api.attributes.BuildTypeAttr=release, com.android.build.gradle.internal.attributes.VariantAttr=release, dexing-enable-desugaring=true, dexing-enable-jacoco-instrumentation=false, dexing-is-debuggable=false, dexing-min-sdk=22, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
         > No variants of com.google.games:gpgs-plugin-support:0.11.01 match the consumer attributes:
             - com.google.games:gpgs-plugin-support:0.11.01 configuration runtime declares a runtime of a component:
                 - Incompatible because this component declares a component, as well as attribute 'artifactType' with value 'srcaar' and the consumer needed a component, as well as attribute 'artifactType' with value 'android-classes-jar'
                 - Other compatible attributes:
                     - Doesn't say anything about asm-transformed-variant (required 'NONE')
                     - Doesn't say anything about com.android.build.api.attributes.AgpVersionAttr (required '7.4.2')
                     - Doesn't say anything about com.android.build.api.attributes.BuildTypeAttr (required 'release')
                     - Doesn't say anything about dexing-enable-desugaring (required 'true')
                     - Doesn't say anything about dexing-enable-jacoco-instrumentation (required 'false')
                     - Doesn't say anything about dexing-is-debuggable (required 'false')
                     - Doesn't say anything about dexing-min-sdk (required '22')
                     - Doesn't say anything about its target Java environment (preferred optimized for Android)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 26s

UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)

```


```
CommandInvokationFailure: Gradle build failed. 
D:\Tool\Unity\2022.3.51f1\Editor\Data\PlaybackEngines\AndroidPlayer\OpenJDK\bin\java.exe -classpath "D:\Tool\Unity\2022.3.51f1\Editor\Data\PlaybackEngines\AndroidPlayer\Tools\gradle\lib\gradle-launcher-7.5.1.jar" org.gradle.launcher.GradleMain "-Dorg.gradle.jvmargs=-Xmx4096m" "assembleRelease"

Environment Variables:
～

stderr[
����:D:\ApplicationUnity_SDK\Google\_GooglePlayGamesPlugin\GPG_0_11_01_U2022_3_51\Library\Bee\Android\Prj\IL2CPP\Gradle\unityLibrary\src\main\java\com\unity3d\player\UnityPlayerActivity.java�͐�������Ȃ�API��g�p�܂��̓I�[�o�[���C�h���Ă��܂��B
����:�ڍׂ́A-Xlint:deprecation�I�v�V������w�肵�čăR���p�C�����Ă��������B

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':launcher:mergeDexRelease'.
> Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
   > Failed to transform classes.jar (project :unityLibrary) to match attributes {artifactType=android-dex, asm-transformed-variant=NONE, com.android.build.api.attributes.AgpVersionAttr=7.4.2, com.android.build.api.attributes.BuildTypeAttr=release, com.android.build.gradle.internal.attributes.VariantAttr=release, dexing-enable-desugaring=true, dexing-enable-jacoco-instrumentation=false, dexing-is-debuggable=false, dexing-min-sdk=22, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
         > No variants of com.google.games:gpgs-plugin-support:0.11.01 match the consumer attributes:
             - com.google.games:gpgs-plugin-support:0.11.01 configuration runtime declares a runtime of a component:
                 - Incompatible because this component declares a component, as well as attribute 'artifactType' with value 'srcaar' and the consumer needed a component, as well as attribute 'artifactType' with value 'android-classes-jar'
                 - Other compatible attributes:
                     - Doesn't say anything about asm-transformed-variant (required 'NONE')
                     - Doesn't say anything about com.android.build.api.attributes.AgpVersionAttr (required '7.4.2')
                     - Doesn't say anything about com.android.build.api.attributes.BuildTypeAttr (required 'release')
                     - Doesn't say anything about dexing-enable-desugaring (required 'true')
                     - Doesn't say anything about dexing-enable-jacoco-instrumentation (required 'false')
                     - Doesn't say anything about dexing-is-debuggable (required 'false')
                     - Doesn't say anything about dexing-min-sdk (required '22')
                     - Doesn't say anything about its target Java environment (preferred optimized for Android)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 26s
]
stdout[

> Configure project :launcher
WARNING:The option setting 'android.aapt2FromMavenOverride=D:\Tool\Unity\2022.3.51f1\Editor\Data\PlaybackEngines\AndroidPlayer\SDK\build-tools\34.0.0\aapt2.exe' is experimental.

> Configure project :unityLibrary:GooglePlayGamesManifest.androidlib
WARNING:minSdkVersion (22) is greater than targetSdkVersion (16) for variant "debug". Please change the values such that minSdkVersion is less than or equal to targetSdkVersion.
WARNING:minSdkVersion (22) is greater than targetSdkVersion (16) for variant "release". Please change the values such that minSdkVersion is less than or equal to targetSdkVersion.
WARNING:We recommend using a newer Android Gradle plugin to use compileSdk = 35

This Android Gradle plugin (7.4.2) was tested up to compileSdk = 33

This warning can be suppressed by adding
    android.suppressUnsupportedCompileSdk=35
to this project's gradle.properties

The build will continue, but you are strongly encouraged to update your project to
use a newer Android Gradle Plugin that has been tested with compileSdk = 35

> Task :unityLibrary:preBuild UP-TO-DATE

```

■About GooglePlayGamesPluginDependencies.xml
The following warning occurs.
Assets/GooglePlayGames/com.google.play.games/Editor/GooglePlayGamesPluginDependencies.xml:11: Repo path 'Packages/com.google.play.games/Editor/m2repository' does not exist.
UnityEngine.Debug:LogWarning (object)

*Assets/GeneratedLocalRepo is not created.

GooglePlayGamesPluginDependencies.xml is set as <repository>Packages/com.google.play.games/Editor/m2repository</repository>.
However, there is no [com.google.play.games/Editor/m2repository] folder in the Packages folder.
Is this specified correctly?
Or are the following settings correct?
<repository>Assets/GooglePlayGames/com.google.play.games/Editor/m2repository</repository>

Best Regards
crSudo

I used Unity 2022.3.52f1 and set the Minimum API Level to 24, but an error was output.

1. Create new Project: Launch on [Windows,Mac,Linux].
2. Add GooglePlayGamesPlugin-0.11.01.unitypackage via Assets > Import Package > Custom Package...
3. Delete External Dependency Manager folder under the Assets folder.
4. Add external-dependency-manager-1.2.183.unitypackage via Assets > Import Package > Custom Package...
5. Unity MenuBar [File]-[Build Settings] Select Android.
6. Select [Switch Platform].
7. [Enable Android Auto-resolution]: Select Enable.
8. The [Google Version Handler] dialog will appear. Select Apply.
9. Select [File]-[Build Settings]-[Player Settings] and change the Minimum API Level to [Android 7.0 API Level 24].
10. Run Assets > External Dependency Manager > Android Resolver > Resolve.
11. Run Assets > External Dependency Manager > Android Resolver > Force Resolve.
The following warning will be displayed.

```
Assets/GooglePlayGames/com.google.play.games/Editor/GooglePlayGamesPluginDependencies.xml:11: Repo path 'Packages/com.google.play.games/Editor/m2repository' does not exist.
UnityEngine.Debug:LogWarning (object)
```
Also, Assets/GeneratedLocalRepo is not created.

12. Run [File]-[Build Setting]-[Build]
The following dialog was displayed:

```
Bulld failure
Could not resolve all files for configuration ':unityLibrary:releaseCompileClasspath'. See the Console for details.
```
When I check the console, the output is as follows:

```
> Configure project :launcher
WARNING:The option setting 'android.aapt2FromMavenOverride=D:\Tool\Unity\2022.3.52f1\Editor\Data\PlaybackEngines\AndroidPlayer\SDK\build-tools\34.0.0\aapt2.exe' is experimental.

> Configure project :unityLibrary:GooglePlayGamesManifest.androidlib
WARNING:minSdkVersion (24) is greater than targetSdkVersion (16) for variant "debug". Please change the values such that minSdkVersion is less than or equal to targetSdkVersion.
WARNING:minSdkVersion (24) is greater than targetSdkVersion (16) for variant "release". Please change the values such that minSdkVersion is less than or equal to targetSdkVersion.
WARNING:We recommend using a newer Android Gradle plugin to use compileSdk = 35

This Android Gradle plugin (7.4.2) was tested up to compileSdk = 33

This warning can be suppressed by adding
    android.suppressUnsupportedCompileSdk=35
to this project's gradle.properties

The build will continue, but you are strongly encouraged to update your project to
use a newer Android Gradle Plugin that has been tested with compileSdk = 35

> Task :unityLibrary:preBuild UP-TO-DATE
～～～～
> Task :unityLibrary:GooglePlayGamesManifest.androidlib:processReleaseManifest
package="com.google.example.games.mainlibproj" found in source AndroidManifest.xml: D:\ApplicationUnity_SDK\Google\_GooglePlayGamesPlugin\GPG_0_11_01_U2022_3_52\Library\Bee\Android\Prj\Mono2x\Gradle\unityLibrary\GooglePlayGamesManifest.androidlib\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.

～～～～
> Task :unityLibrary:GooglePlayGamesManifest.androidlib:compileReleaseJavaWithJavac
54 actionable tasks: 54 executed

UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)
```

```
FAILURE: Build completed with 3 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':unityLibrary:generateReleaseRFile'.
> Could not resolve all files for configuration ':unityLibrary:releaseCompileClasspath'.
   > Could not find com.google.games:gpgs-plugin-support:0.11.01.
     Required by:
         project :unityLibrary

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

3: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':launcher:checkReleaseDuplicateClasses'.
> Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
   > Could not find com.google.games:gpgs-plugin-support:0.11.01.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - https://repo.maven.apache.org/maven2/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GPG_0_11_01_U2022_3_52/Packages/com.google.play.games/Editor/m2repository/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/C:/Users/nor25/.m2/repository/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GPG_0_11_01_U2022_3_52/Library/Bee/Android/Prj/Mono2x/Gradle/unityLibrary/libs/gpgs-plugin-support-0.11.01.jar
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GPG_0_11_01_U2022_3_52/Library/Bee/Android/Prj/Mono2x/Gradle/unityLibrary/libs/gpgs-plugin-support.jar
     Required by:
         project :launcher > project :unityLibrary

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 8s

UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)
```

```
CommandInvokationFailure: Gradle build failed. 
D:\Tool\Unity\2022.3.52f1\Editor\Data\PlaybackEngines\AndroidPlayer\OpenJDK\bin\java.exe -classpath "D:\Tool\Unity\2022.3.52f1\Editor\Data\PlaybackEngines\AndroidPlayer\Tools\gradle\lib\gradle-launcher-7.5.1.jar" org.gradle.launcher.GradleMain "-Dorg.gradle.jvmargs=-Xmx4096m" "assembleRelease"

Environment Variables:
～～～～

stderr[

FAILURE: Build completed with 3 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':unityLibrary:generateReleaseRFile'.
> Could not resolve all files for configuration ':unityLibrary:releaseCompileClasspath'.
   > Could not find com.google.games:gpgs-plugin-support:0.11.01.
     Required by:
         project :unityLibrary

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

3: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':launcher:checkReleaseDuplicateClasses'.
> Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
   > Could not find com.google.games:gpgs-plugin-support:0.11.01.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - https://repo.maven.apache.org/maven2/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GPG_0_11_01_U2022_3_52/Packages/com.google.play.games/Editor/m2repository/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/C:/Users/nor25/.m2/repository/com/google/games/gpgs-plugin-support/0.11.01/gpgs-plugin-support-0.11.01.pom
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GPG_0_11_01_U2022_3_52/Library/Bee/Android/Prj/Mono2x/Gradle/unityLibrary/libs/gpgs-plugin-support-0.11.01.jar
       - file:/D:/ApplicationUnity_SDK/Google/_GooglePlayGamesPlugin/GPG_0_11_01_U2022_3_52/Library/Bee/Android/Prj/Mono2x/Gradle/unityLibrary/libs/gpgs-plugin-support.jar
     Required by:
         project :launcher > project :unityLibrary

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 8s
]
stdout[

> Configure project :launcher
WARNING:The option setting 'android.aapt2FromMavenOverride=D:\Tool\Unity\2022.3.52f1\Editor\Data\PlaybackEngines\AndroidPlayer\SDK\build-tools\34.0.0\aapt2.exe' is experimental.

> Configure project :unityLibrary:GooglePlayGamesManifest.androidlib
WARNING:minSdkVersion (24) is greater than targetSdkVersion (16) for variant "debug". Please change the values such that minSdkVersion is less than or equal to targetSdkVersion.
WARNING:minSdkVersion (24) is greater than targetSdkVersion (16) for variant "release". Please change the values such that minSdkVersion is less than or equal to targetSdkVersion.
WARNING:We recommend using a newer Android Gradle plugin to use compileSdk = 35

This Android Gradle plugin (7.4.2) was tested up to compileSdk = 33

This warning can be suppressed by adding
    android.suppressUnsupportedCompileSdk=35
to this project's gradle.properties

The build will continue, but you are strongly encouraged to update your project to
use a newer Android Gradle Plugin that has been tested with compileSdk = 35

> Task :unityLibrary:preBuild UP-TO-DATE
～～～～
> Task :unityLibrary:GooglePlayGamesManifest.androidlib:processReleaseManifest
package="com.google.example.games.mainlibproj" found in source AndroidManifest.xml: D:\ApplicationUnity_SDK\Google\_GooglePlayGamesPlugin\GPG_0_11_01_U2022_3_52\Library\Bee\Android\Prj\Mono2x\Gradle\unityLibrary\GooglePlayGamesManifest.androidlib\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.

～～～～
> Task :unityLibrary:GooglePlayGamesManifest.androidlib:compileReleaseJavaWithJavac
54 actionable tasks: 54 executed
]
exit code: 1
UnityEditor.Android.Command.WaitForProgramToRun (UnityEditor.Utils.Program p, UnityEditor.Android.Command+WaitingForProcessToExit waitingForProcessToExit, System.String errorMsg) (at <681810dba5f047c6addcc3cde9f9ce21>:0)
UnityEditor.Android.Command.Run (System.Diagnostics.ProcessStartInfo psi, UnityEditor.Android.Command+WaitingForProcessToExit waitingForProcessToExit, System.String errorMsg) (at <681810dba5f047c6addcc3cde9f9ce21>:0)
UnityEditor.Android.Command.Run (System.String command, System.String args, System.String workingdir, UnityEditor.Android.Command+WaitingForProcessToExit waitingForProcessToExit, System.String errorMsg) (at <681810dba5f047c6addcc3cde9f9ce21>:0)
UnityEditor.Android.AndroidJavaTools.RunJava (System.String args, System.String workingdir, System.Action`1[T] progress, System.String error) (at <681810dba5f047c6addcc3cde9f9ce21>:0)
UnityEditor.Android.GradleWrapper.Run (UnityEditor.Android.AndroidJavaTools javaTools, Unity.Android.Gradle.AndroidGradle androidGradle, System.String workingdir, System.String task, System.Action`1[T] progress) (at <681810dba5f047c6addcc3cde9f9ce21>:0)
Rethrow as GradleInvokationException: Gradle build failed
UnityEditor.Android.GradleWrapper.Run (UnityEditor.Android.AndroidJavaTools javaTools, Unity.Android.Gradle.AndroidGradle androidGradle, System.String workingdir, System.String task, System.Action`1[T] progress) (at <681810dba5f047c6addcc3cde9f9ce21>:0)
UnityEditor.Android.PostProcessor.Tasks.BuildGradleProject.Execute (UnityEditor.Android.PostProcessor.PostProcessorContext context) (at <681810dba5f047c6addcc3cde9f9ce21>:0)
UnityEditor.Android.PostProcessor.PostProcessRunner.RunAllTasks (UnityEditor.Android.PostProcessor.PostProcessorContext context) (at <681810dba5f047c6addcc3cde9f9ce21>:0)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr, Boolean&)
```

```
Build completed with a result of 'Failed' in 47 seconds (46513 ms)
3 errors
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)
```

Best Regards
crSudo

@argzdev 

Thank you.
After correcting GooglePlayGamesPluginDependencies.xml, I was able to build correctly.
However, when I used the sample code, pressed the [Achievements]-[Ach ShowUI] button and executed Social.ShowAchievementsUI(), the following error occurred.
\play-games-plugin-for-unity-0.11.01\play-games-plugin-for-unity-0.11.01\Samples\SmokeTest

```
E  AndroidJavaException: java.lang.ClassNotFoundException: com.google.games.bridge.HelperFragment
	java.lang.ClassNotFoundException: com.google.games.bridge.HelperFragment
		at java.lang.Class.classForName(Native Method)
		at java.lang.Class.forName(Class.java:607)
		at com.unity3d.player.UnityPlayer.nativeRender(Native Method)
		at com.unity3d.player.UnityPlayer.-$$Nest$mnativeRender(Unknown Source:0)
		at com.unity3d.player.G0.handleMessage(Unknown Source:122)
		at android.os.Handler.dispatchMessage(Handler.java:102)
		at android.os.Looper.loopOnce(Looper.java:205)
		at android.os.Looper.loop(Looper.java:294)
		at com.unity3d.player.H0.run(Unknown Source:24)
	Caused by: java.lang.ClassNotFoundException: com.google.games.bridge.HelperFragment
		at java.lang.Class.classForName(Native Method) 
		at java.lang.Class.forName(Class.java:607) 
		at com.unity3d.player.UnityPlayer.nativeRender(Native Method) 
		at com.unity3d.player.UnityPlayer.-$$Nest$mnativeRender(Unknown Source:0) 
		at com.unity3d.player.G0.handleMessage(Unknown Source:122) 
		at android.os.Handler.dispatchMessage(Handler.java:102) 
		at android.os.Looper.loopOnce(Looper.java:205) 
		at android.os.Looper.loop(Looper.java:294) 
	at com.unity3d.player.H0.run(Unknown Source:24) 
	at UnityEngine.AndroidJNISafe.CheckException () [0x00000] in <00000000000000000000000000000000>:0 
	at UnityEngine.AndroidJNISafe.FindClass (System.String name) [0x00000] in <00000000000000000000000000000000>:0 
	at UnityEngine.AndroidJavaClass._AndroidJavaClass (Sys
I   [Play Games Plugin 0.11.01] 11/11/24 20:12:01 +09:00 DEBUG: ShowAchievementsUI callback is 
   GooglePlayGames.OurUtils.PlayGamesHelperObject:Update()
E  [17949]-[SurfaceView[jp.co.ofcr.cm00/com.unity3d.player.UnityPlayerActivity](BLAST)#17949]-timeRecords is at its maximum size[64]. Ignore this when unittesting.

```
I tried using external-dependency-manager-1.2.182.unitypackage and it worked correctly.

The environment is:
・Unity 2022.3.52f1
・GooglePlayGamesPlugin-0.11.01.unitypackage
・external-dependency-manager-1.2.183.unitypackage *Error occurred
・external-dependency-manager-1.2.182.unitypackage *Works normally.

Best Regards
crSudo

Thank you for your research, @argzdev.

When I set the data in [Resources Definition] in the Unity menu bar [Windows]-[Google Play Games]-[Setup]-[Android setup], GooglePlayGamesPluginDependencies.xml was automatically corrected and the error disappeared.
However, when I run the Sample [Achievements]-[Ach ShowUI], the following error occurs.

```
SmokeTest.AchievementGUI:OnGUI()
AndroidJavaException: java.lang.ClassNotFoundException: com.google.games.bridge.HelperFragment
java.lang.ClassNotFoundException: com.google.games.bridge.HelperFragment
	at java.lang.Class.classForName(Native Method)
	at java.lang.Class.forName(Class.java:607)
	at com.unity3d.player.UnityPlayer.nativeRender(Native Method)
	at com.unity3d.player.UnityPlayer.-$$Nest$mnativeRender(Unknown Source:0)
	at com.unity3d.player.G0.handleMessage(Unknown Source:122)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at com.unity3d.player.H0.run(Unknown Source:24)
Caused by: java.lang.ClassNotFoundException: com.google.games.bridge.HelperFragment
	at java.lang.Class.classForName(Native Method) 
	at java.lang.Class.forName(Class.java:607) 
	at com.unity3d.player.UnityPlayer.nativeRender(Native Method) 
	at com.unity3d.player.UnityPlayer.-$$Nest$mnativeRender(Unknown Source:0) 
	at com.unity3d.player.G0.handleMessage(Unknown Source:122) 
	at android.os.Handler.dispatchMessage(Handler.java:102) 
	at android.os.Looper.loopOnce(Looper.java:205) 
	at android.os.Looper.loop(Looper.java:294) 
	at com.unity3d.player.H0.run(Unknown Source:24) 
  at UnityEngine.AndroidJNISafe.CheckException () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.AndroidJNISafe.FindClass (System.String name) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.AndroidJavaClass._AndroidJavaClass (Sys
 [Play Games Plugin 0.11.01] 11/16/24 18:06:41 +09:00 DEBUG: ShowAchievementsUI callback is 
GooglePlayGames.OurUtils.PlayGamesHelperObject:Update()

```
The steps are as follows:
1. Create new Project: Launch on [Windows,Mac,Linux].
2. Add GooglePlayGamesPlugin-0.11.01.unitypackage via Assets > Import Package > Custom Package...
3. Delete External Dependency Manager folder under the Assets folder.
4. Add external-dependency-manager-1.2.183.unitypackage via Assets > Import Package > Custom Package...
5. Unity MenuBar [File]-[Build Settings] Select Android.
6. Select [Switch Platform].
7. [Enable Android Auto-resolution]: Select Enable.
8. The [Google Version Handler] dialog will appear. Select Apply.
9. Unity MenuBar [Windows]-[Google Play Games]-[Setup]-[Android setup]:[Resources Definition] settings
10. [Set Bundle Identifier] dialog will appear. Select OK.
11. Select Unity MenuBar [File]-[Build Settings]-[Player Settings]
    [Orther Settings]
        [Minimum API Level] : [Android 7.0 API Level 24].
        [Scripting Backend] : IL2CPP
        [Target Architectures]: Check ARMv64
    [Publish Settings]
        I set [Project Keystore] and [Project Key].
12. Run Assets > External Dependency Manager > Android Resolver > Resolve.
13. Run Assets > External Dependency Manager > Android Resolver > Force Resolve.
14. Run [File]-[Build Setting]-[Build]

However, if you manually change the path in \Assets\Plugins\Android\settingsTemplate.gradle as follows, it seems to work correctly.
url (unityProjectPath + "/Assets/GeneratedLocalRepo/GooglePlayGames/com.google.play.games/Editor/m2repository")

Best Regards
crSudo

--- a/source/AndroidResolver/src/GradleTemplateResolver.cs
+++ b/source/AndroidResolver/src/GradleTemplateResolver.cs
@@ -730,9 +730,12 @@ namespace GooglePlayServices {
                         // repo path contains .srcaar.
                         var repoPath = FileUtils.ReplaceBaseAssetsOrPackagesFolder(
                                 relativePath, GooglePlayServices.SettingsDialog.LocalMavenRepoDir);
-                        // We also want to just convert any prefixes before a directory/m2repository, since
-                        // they are copied to the LocalMavenRepoDir as well.
-                        repoPath = ReplaceLocalFolderBasedOnM2repo(repoPath);
+
+                        if (!repoPath.StartsWith(GooglePlayServices.SettingsDialog.LocalMavenRepoDir)) {
+                            // same replacement logic as in CopySrcAars() 
+                            repoPath = ReplaceLocalFolderBasedOnM2repo(repoPath);
+                        }
+                        
                         if (!Directory.Exists(repoPath)) {
                             repoPath = relativePath;
                         }

